src/lib/lib_weechat.py

```diff
@@ -51,9 +51,13 @@ def weechat_command(self, command):
             return 'Empty command'
         cmd = "*{}".format(command)
         hookenv.log("Calling fifo with: {}".format(cmd))
-        result = subprocess.check_output('echo {} > {}'.format(cmd, self.fifo_file),
-                                         shell=True)
-        return result.decode()
+        try:
+            with open(self.fifo_file,"w") as fifo:
+                print(cmd, file = fifo)
+        except:
```

Comment:
I've violated it myself before as well, but a bare except like this isn't a great idea. Also if this does have an issue, maybe a hookenv.log for ERROR should be provided?